covid19-dashboard/dc-covid19/src/App.tsx

```diff
@@ -14,186 +14,122 @@
-                        Object.keys(json).forEach(geoId => {
-                            if (json[geoId][1] === 'country/USA') regions[geoId] = json[geoId][0]
-                        })
-                        this.setState({availableRegions: {...this.state.availableRegions, ...regions}})
-                    }
-                })
-            })
-        })
-    }
-
-    /**
-     * Creates a reference for each row which allows easy access from the SideNav.
-     */
-    generateReferences = () => {
-        let refs_ = {}
-        for (let key in ContentFile) refs_[key] = React.createRef<HTMLDivElement>()
-        return refs_
-    }
-
-    /**
-     * In charge of scrolling to a specific reference in the page.
-     * @param event
-     */
-    handleScrollOnRef = (event) => {
-        const ref_ = this.refs_[event.target.id]
-        if (ref_.current) {
-            ref_.current.scrollIntoView({
-                behavior: "smooth",
-                block: "nearest"
-            })
-        }
-    }
-
-    /**
-     * In charge of changing the state when the region, date or showTopN are updated.
-     * @param key
-     * @param value
-     */
-    handleSelectUpdate = (key: string, value: any) =>{
-        switch(key) {
-            case 'region':
-                this.setState({selectedRegion: value})
-                break;
-            case 'date':
-                this.setState({datePicked: value})
-                break;
-            case 'showTopN':
-                this.setState({selectedShowTopN: value})
-                break;
-        }
-    }
+/**
+ * Given a date and an array of deltaDays compute the real ISODates.
+ * For example, date = 2020-01-10 and deltaDays = [0, 1, 7] will return an array with the ISODates
+ * [2020-01-10, 2020-01-09, 2020-01-03] because 0 represents today, 1 represents 1 day ago, and 7 represents 7 days ago.
+ * @param date
+ * @param deltaDays
+ */
+const getRealISODatesFromArrayOfDeltaDays = (
+  date: string,
+  deltaDays: number[]
+): string[] => {
+  if (date === undefined || date === null) return [];
+  const dates = deltaDays.map(deltaDate =>
+    addOrSubtractNDaysToDate(date, -deltaDate)
+  );
+  return dates.sort((a, b) => (a < b ? -1 : a > b ? 1 : 0));
+};
 
-    /**
-     * Request the data from the server, and store the Promises in that order.
-     * The order is important [cases, deaths, population, places].
-     */
-    fetchData = () => {
-        const url: string = `/api/`
-        const apis = [
-            url + 'total-cases',
-            url + 'total-deaths',
-            url + 'population',
-            url + 'places'];
-        // Map every url to the promise of the fetch
-        return apis.map(api => fetch(api));
-    }
+/**
+ * Converts a number to a string including commas for readability.
+ * For example, int(10000) would get converted to str(10,000)
+ * @param num: the number to replace
+ */
+const numberWithCommas = (num: number): string => {
```

Comment:
how about using this library: https://www.npmjs.com/package/human-readable-numbers